extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationService.js

```diff
@@ -148,6 +148,35 @@ oppia.factory('ItemSelectionInputValidationService', [
             return handledAnswer;
           });
         }
+        var answerChoiceToIndex = {};
+        seenChoices.forEach(function(seenChoice, choiceIndex) {
```

Comment:
Before doing this, have we confirmed that all choices are distinct? Otherwise you may overwrite.

If not, perhaps that seems like something we should confirm.

Reply:
I will check it after 9, (4 hr from now), have an assignment deadline, ill quickly get done with it and check. 
Thanks! and sorry for delay

Reply:
hi! @seanlip 
I am not sure but I do not think we will be overwriting It in this case, I will only mark handled answers when I find them with the given rule. 

Also I apologize for not getting back, I have no idea how I missed it. I will be more careful.

Reply:
Let me be clearer -- suppose choice 1 is "aaa" and choice 2 is also "aaa". When we update in the for loop, wouldn't we be overwriting "aaa" by assigning it to index 2 (after it has already been assigned to index 1)?

(Also, no worries about the delay. As long as you make a note here stating clearly when you'll get back to it, that is fine.)

Reply:
The choices are distinct, I found this part of code that checks this ,  (in the validation) 
```javascript
      for (var i = 0; i < numChoices; i++) {
          var choice = customizationArgs.choices.value[i];
          if (choice.trim().length === 0) {
            areAnyChoicesEmpty = true;
          }
          if (seenChoices.indexOf(choice) !== -1) {
            areAnyChoicesDuplicated = true;
          }
          seenChoices.push(choice);
          handledAnswers.push(false);
     }
```
```javascript
        if (areAnyChoicesDuplicated) {
          warningsList.push({
            type: WARNING_TYPES.CRITICAL,
            message: 'Please ensure the choices are unique.'
          });
        }
```
So I dont think we will face choice1 = choice 2 anytime.  So that way, they will not be overwritten

Reply:
Yeah sorry that slipped through, I did not check that, It is being erased and written again to the exact same thing, thats how It passed all the tests, So basically I think it is one loop running two times, I will remove it now